### ISSUE TITLE
Stance on GNAP Adoption

### ISSUE THREAD

**Issue by matthew-salerno** [kanidm/kanidm#3045]
I've just started reading the gnap Internet-Draft and am very interested in it's development. I was wondering if Kanidm had plans to support the protocol once it became a proper spec.

I'm not expecting support for an Internet Draft, I'm just curious about it and asked on Matrix if the Kanidm devs have any thoughts or plans around it in the future. But Matrix isn't searchable so I'm asking here for posterity, since I imagine others might be interested.

**Comment by yaleman** [comment]
https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol for reference

once it's less drafty, maybe...

**Comment by Firstyear** [comment]
Had a quick read - looks like an "improved" oauth2 / oidc framework, but would sit in parallel. I'm not against it at all, but as mentioned I'd like to see it finalised as a spec, and there to be some compelling reasons to adopt it outside of oauth2. Thanks for letting us know about it though!

**Comment by matthew-salerno** [comment]
You're probobly already further ahead than I am in it :) The main arguments I've seen from GNAP proponents is:
1. Doesn't assume browsers as the method of Auth with the Authorization Server
2. More narrow implementation than OAuth, I'm sure it's easy to forget once you figure OAuth/OIDC out, but OAuth is confusing to learn with all its optional configuration and extensions. See: [it's time for OAuth 2.1](https://aaronparecki.com/2019/12/12/21/its-time-for-oauth-2-dot-1).
3. Client ID's can be received dynamically, which opens the doors for things like 3rd party email clients.

I am of course not an expert, just wanted to put it on the radar. Appendix B covers differences better than I can.

**Comment by Firstyear** [comment]
Yeah, I think there is a lot in there to take it, it's written by an oauth2 spec contributor so they seem to be knowledgable about the weaknesses of oauth2 as well.  I think I want to review it again before I really comment more.

**Comment by fipoac** [comment]
I know it's not a priority right now, but GNAP was standardised into [RFC 9635](https://datatracker.ietf.org/doc/html/rfc9635)
